sc2/game_info.py

```diff
@@ -176,59 +176,49 @@ def __init__(self, proto):
 
     def _find_ramps(self) -> List[Ramp]:
         """Calculate (self.pathing_grid - self.placement_grid) (for sets) and then find ramps by comparing heights."""
-        rampPoints = {
+        map_area = self.playable_area
+        rampPoints = [
             Point2((x, y))
-            for x in range(self.pathing_grid.width)
-            for y in range(self.pathing_grid.height)
+            for x in range(map_area.x, map_area.x + map_area.width + 1)
+            for y in range(map_area.y, map_area.y + map_area.height + 1)
             if self.placement_grid[(x, y)] == 0 and self.pathing_grid[(x, y)] == 0
-        }
+        ]
         return [Ramp(group, self) for group in self._find_groups(rampPoints)]
 
-    def _find_groups(
-        self, points: Set[Point2], minimum_points_per_group: int = 8, max_distance_between_points: int = 2
-    ) -> List[Set[Point2]]:
+    def _find_groups(self, points: Set[Point2], minimum_points_per_group: int = 8):
         """
         From a set of points, this function will try to group points together by
         painting clusters of points in a rectangular map using flood fill algorithm.
         Returns groups of points as list, like [{p1, p2, p3}, {p4, p5, p6, p7, p8}]
         """
-        NOT_INTERESTED = -2
+        # TODO do we actually need colors here? the ramps will never touch amyways.
```

Comment:
Should this be resolved first?

Reply:
This is not a huge deal, the total time to run this function is between 0.016 and 0.02 sec on my machine. Also it is only called one time in a game.

Reply:
Ok, this is fine